### PR TITLE
CNI configmap + dnsPolicy for operator in managed ETCd mode.

### PR DESCRIPTION
**Pull request by wizard580**
Bump chart version to 1.8.3
Fixes: 
1. No Helm way to provide CNI config for cilium. So we introduce template to do this.
2. Introduce configurable dnsPolicy when managed ETCd is used.

Description:
1. if we want to customize CNI config we can do this in the following path: `.global.cni.*` after we define `.global.cni.managedConfigmap`. Before that the only way to customize CNI config was to manually create configmap and provide it as param `.global.cni.configMap` for the helm install. Usable in combination with `.global.cni.configMap=cilium-cni-configuration`

This is not breaking change for existing installations.

Supported config params as of now:
`.global.cni.version`
`.global.cni.name`
`.global.cni.type`
`.global.cni.debug`
`.global.cni.eni.first-interface-index`

2. In my case chart v1.8.2 wasn't able to provide functioning Cilium installation on clean EKS cluster if I tried to install Cilium with managed ETCd. So to fix the chicken-egg issue I introducing param `.global.operator.dnsPolicy` which is usable only if we enabled `.global.etcd.managed`. By default `.global.operator.dnsPolicy` is falling back to `ClusterFirstWithHostNet` and for me value `Default` is the fix. 

So this isn't a breaking change for existing installations.

Supported config params as of now:
`.global.operator.dnsPolicy`

### CODE REVIEW

File: install/kubernetes/cilium/charts/config/templates/cni-configmap.yaml
Comment (joestringer) [inline install/kubernetes/cilium/charts/config/templates/cni-configmap.yaml:28]: So far, there has been no option `global.cni.eni`. The EKS/ENI docs use `global.eni`. Is this what you meant?

**Review by joestringer — CHANGES_REQUESTED** [review]
Hi, thanks for the contribution!

Please target the changes against the `master` branch, we will handle backporting (including the chart version bumps) to v1.8 after it has been reviewed & merged into the main branch. 

As the bot has already mentioned, each commit also needs to be signed off ("git commit -s"), for more information see the [Developer's Certificate of Origin](https://docs.cilium.io/en/v1.8/contributing/development/contributing_guide/#developer-s-certificate-of-origin).

I will defer to @cilium/helm team to look over the helm charts in more detail.

File: install/kubernetes/cilium/charts/config/templates/cni-configmap.yaml
Comment (wizard580) [reply install/kubernetes/cilium/charts/config/templates/cni-configmap.yaml:28]: partially.
My goal here was to configure CNI to set index of first used ENI. 
Potentially there may be params other than `first-interface-index`, so I used `global.cni.eni.` as placeholder. And we will be able to use something like this if needed:
```
global:
  cni:
    eni:
      first-interface-index: 0
      blablaparam: 111
```

Comment (wizard580) [reply install/kubernetes/cilium/charts/config/templates/cni-configmap.yaml:28]: No option before - not breaking change.

**Pull request closed without merging**
